iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/CatalogWrapperForREST.java

```diff
@@ -0,0 +1,157 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.gravitino.catalog.lakehouse.iceberg.IcebergConstants;
+import org.apache.gravitino.credential.CatalogCredentialManager;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.credential.CredentialConstants;
+import org.apache.gravitino.credential.CredentialPropertyUtils;
+import org.apache.gravitino.credential.PathBasedCredentialContext;
+import org.apache.gravitino.iceberg.common.IcebergConfig;
+import org.apache.gravitino.iceberg.common.ops.IcebergCatalogWrapper;
+import org.apache.gravitino.utils.MapUtils;
+import org.apache.gravitino.utils.PrincipalUtils;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.ServiceUnavailableException;
+import org.apache.iceberg.rest.requests.CreateTableRequest;
+import org.apache.iceberg.rest.responses.LoadTableResponse;
+
+/** Process Iceberg REST specific operations, like credential vending. */
+public class CatalogWrapperForREST extends IcebergCatalogWrapper {
+
+  private final CatalogCredentialManager catalogCredentialManager;
+
+  private final Map<String, String> catalogConfigToClients;
+
+  private static final Set<String> catalogPropertiesToClientKeys =
+      ImmutableSet.of(
+          IcebergConstants.IO_IMPL,
+          IcebergConstants.AWS_S3_REGION,
+          IcebergConstants.ICEBERG_S3_ENDPOINT,
+          IcebergConstants.ICEBERG_OSS_ENDPOINT);
+
+  public CatalogWrapperForREST(String catalogName, IcebergConfig config) {
+    super(config);
+    this.catalogConfigToClients =
+        MapUtils.getFilteredMap(
+            config.getIcebergCatalogProperties(),
+            key -> catalogPropertiesToClientKeys.contains(key));
+    // To compatibility with old version
+    Map<String, String> catalogProperties = normalizeCredentialProperties(config.getAllConfig());
+    this.catalogCredentialManager = new CatalogCredentialManager(catalogName, catalogProperties);
+  }
+
+  public LoadTableResponse createTable(
+      Namespace namespace, CreateTableRequest request, boolean requestCredential) {
+    LoadTableResponse loadTableResponse = super.createTable(namespace, request);
+    if (requestCredential) {
+      return injectCredentialConfig(
+          TableIdentifier.of(namespace, request.name()), loadTableResponse);
+    }
+    return loadTableResponse;
+  }
+
+  public LoadTableResponse loadTable(TableIdentifier identifier, boolean requestCredential) {
+    LoadTableResponse loadTableResponse = super.loadTable(identifier);
+    if (requestCredential) {
+      return injectCredentialConfig(identifier, loadTableResponse);
+    }
+    return loadTableResponse;
+  }
```

Comment:
I assume these two methods should add `@Override` annotation, right?

Reply:
no, the method is different for `requestCredential` parameter